src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/AlwaysEncryptedEnclaveProviderUtils.cs

```diff
@@ -18,23 +18,21 @@ public EnclavePublicKey(byte[] payload)
 
     internal class EnclaveDiffieHellmanInfo
     {
-        public int Size { get; private set; }
+        public int Size => sizeof(int) + sizeof(int) + PublicKey?.Length ?? 0 + PublicKeySignature?.Length ?? 0;
```

Comment:
I'm curious how did Size = `payload.Length` from the ctor end up being this calculation.

Reply:
I think it was a knock on effect of the size calculation being incorrect elsewhere. 

It feels like the author had a problem with being a few bytes off and was using Size to help try and work out where it was, once they found or resolved it they didn't feel the need to go back and change them all around. I changed it so that rather than keeping the number of bytes that were made available we report the actual number of bytes that were read.